src/wallet/wallet.cpp

```diff
@@ -5365,39 +5019,69 @@ bool CWallet::CreateZerocoinSpendTransaction(std::string &thirdPartyaddress, int
     return true;
 }
 
-bool CWallet::CreateSigmaSpendTransaction(
-        std::string &thirdPartyaddress,
-        sigma::CoinDenomination denomination,
-        CWalletTx &wtxNew, CReserveKey &reservekey,
-        CAmount& nFeeRet, Scalar &coinSerial,
-        uint256 &txHash, GroupElement &zcSelectedValue, bool &zcSelectedIsUsed,
-        std::string &strFailReason,  bool forceUsed,
-        const CCoinControl *coinControl) {
-
+CWalletTx CWallet::CreateSigmaSpendTransaction(
+    const std::vector<CRecipient>& recipients,
+    CAmount& fee,
+    std::vector<CSigmaEntry>& selected,
+    std::vector<CHDMint>& changes,
+    bool& fChangeAddedToFee,
+    const CCoinControl *coinControl)
+{
+    int nHeight = chainActive.Height();
+    if(nHeight >= ::Params().GetConsensus().nDisableUnpaddedSigmaBlock && nHeight < ::Params().GetConsensus().nSigmaPaddingBlock)
+        throw std::runtime_error(_("Sigma is disabled at this period."));
```

Comment:
This checks are not relevant currently, should be removed and padding flag should be always set to true. There are creation and mempool-related checks not covered by this PR which should be removed too (`validation.cpp:3471`, `sigmaspendbuilder.cpp:76`, `rpctx.cpp:1757` and probably others). Only `CheckTransaction` when transaction is mined into the block should care about non-true padding flag